internal/k8s/controllers/config_conversion_test.go

```diff
@@ -0,0 +1,235 @@
+				BFDProfile: "default",
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "peer2",
+				Namespace: "metallb-system",
+			},
+			Spec: v1beta2.BGPPeerSpec{
+				MyASN:      42,
+				ASN:        142,
+				Address:    "1.2.3.5",
+				BFDProfile: "default",
+			},
+		},
+	}
+	bfdProfiles := []v1beta1.BFDProfile{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "default",
+				Namespace: "metallb-system",
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "bar",
+				Namespace: "metallb-system",
+			},
+		},
+	}
+	pools := []v1beta1.IPAddressPool{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "pool1",
+				Namespace: "metallb-system",
+			},
+			Spec: v1beta1.IPAddressPoolSpec{
+				Addresses: []string{
+					"10.20.0.0/16",
+				},
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "pool2",
+				Namespace: "metallb-system",
+			},
+			Spec: v1beta1.IPAddressPoolSpec{
+				Addresses: []string{
+					"10.10.0.0/16",
+				},
+				AllocateTo: &v1beta1.ServiceAllocation{
+					Namespaces: []string{"foo", "bar"},
+				},
+			},
+		},
+	}
+	bgpAdvs := []v1beta1.BGPAdvertisement{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "adv1",
+				Namespace: "metallb-system",
+			},
+			Spec: v1beta1.BGPAdvertisementSpec{
+				Communities: []string{"bar"},
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "adv2",
+				Namespace: "metallb-system",
+			},
+			Spec: v1beta1.BGPAdvertisementSpec{
+				Communities: []string{"bar2"},
+			},
+		},
+	}
+	l2Advs := []v1beta1.L2Advertisement{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "l2adv1",
+				Namespace: "metallb-system",
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "l2adv2",
+				Namespace: "metallb-system",
+			},
+			Spec: v1beta1.L2AdvertisementSpec{
+				Interfaces: []string{"foo"},
+			},
+		},
+	}
+
+	communities := []v1beta1.Community{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "community",
+				Namespace: "metallb-system",
+			},
+			Spec: v1beta1.CommunitySpec{
+				Communities: []v1beta1.CommunityAlias{
+					{
+						Name:  "bar",
+						Value: "64512:1234",
+					},
+				},
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "community1",
+				Namespace: "metallb-system",
+			},
+			Spec: v1beta1.CommunitySpec{
+				Communities: []v1beta1.CommunityAlias{
+					{
+						Name:  "bar2",
+						Value: "64512:1235",
+					},
+				},
+			},
+		},
+	}
+	namespaces := []v1.Namespace{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "foo",
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "bar",
+			},
+		},
+	}
+	nodes := []v1.Node{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "node1",
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "node2",
+			},
+		},
+	}
+
+	oldConfig, err := toConfig(config.DontValidate,
+		pools,
+		peers,
+		bfdProfiles,
+		l2Advs,
+		bgpAdvs,
+		[]v1beta1.AddressPool{},
```

Comment:
Why do we have the `AddressPool` empty here?
We do run sortedCopy against it in `toConfig` I think.

Reply:
adding

Reply:
done